### PR TITLE
Switch to shared gRPC server implementation

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Fixes https://github.com/crossplane/terrajet/issues/261

Switches to the shared gRPC server implementation in `provider-jet-azure`.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested using the corresponding Terrajet PR: https://github.com/crossplane/terrajet/pull/267

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -27,15 +27,16 @@ import (
...
-	"github.com/crossplane/terrajet/pkg/terraform"
 	"gopkg.in/alecthomas/kingpin.v2"
 	kerrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/tools/leaderelection/resourcelock"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
+	tjcontroller "github.com/crossplane/terrajet/pkg/controller"
+	"github.com/crossplane/terrajet/pkg/terraform"
```
Comment (muvaf) [inline cmd/provider/main.go:38]: I think the crossplane-runtime imports should be moved to this group.

File: internal/clients/azure.go [hunk 2]
```diff
@@ -107,14 +100,11 @@ func spAuth(ctx context.Context, pc *v1alpha1.ProviderConfig, ps *terraform.Setu
...
+	// set credentials configuration
 	ps.Configuration[keySubscriptionID] = azureCreds[keyAzureSubscriptionID]
-	// set credentials environment
-	ps.Env = []string{
-		fmt.Sprintf(fmtEnvVar, envSubscriptionID, azureCreds[keyAzureSubscriptionID]),
-		fmt.Sprintf(fmtEnvVar, envTenantID, azureCreds[keyAzureTenantID]),
-		fmt.Sprintf(fmtEnvVar, envClientID, azureCreds[keyAzureClientID]),
-		fmt.Sprintf(fmtEnvVar, envClientSecret, azureCreds[keyAzureClientSecret]),
-	}
+	ps.Configuration[keyTenantID] = azureCreds[keyAzureTenantID]
```
Comment (muvaf) [inline internal/clients/azure.go:105]: Could you confirm that these are not printed in the error logs?

File: cluster/images/provider-jet-azure-controller/Dockerfile [hunk 3]
```diff
@@ -5,29 +5,32 @@ ARG ARCH
 ARG TERRAFORM_VERSION
 ARG TERRAFORM_PROVIDER_SOURCE
 ARG TERRAFORM_PROVIDER_VERSION
-ARG TINI_VERSION
```
Comment (muvaf) [inline cluster/images/provider-jet-azure-controller/Dockerfile:8]: Was this a remnant of something else?

**Review by muvaf — APPROVED** [review]
LGTM! Thanks @ulucinar !

File: cluster/images/provider-jet-azure-controller/Dockerfile (hunk 3 above)
Comment (ulucinar) [reply cluster/images/provider-jet-azure-controller/Dockerfile:8]: Yes, looks like it's no longer used.

File: internal/clients/azure.go (hunk 2 above)
Comment (ulucinar) [reply internal/clients/azure.go:105]: Hi @muvaf,
I gave this a try by injecting an invalid key in Terraform provider configuration block. Here's what appears in managed resource status:
```yaml
  - message: 'observe failed: cannot run refresh: refresh failed: Extraneous JSON
      object property: No argument or block type is named "non-existent".: File name:
      main.tf.json'
    reason: ReconcileError
    status: "False"
    type: Synced
```

And no relevant provider logs unless debug logging is enabled. If debug logs are enabled, then I think we are dumping Terraform output, which may, in theory leak sensitive data. We can discuss this general issue in Terrajet.

**Pull request merged**